tesseract/consensus/arb-host/src/host.rs

```diff
@@ -13,8 +22,92 @@ impl IsmpHost for ArbHost {
 		&self,
 		counterparty: Arc<dyn IsmpProvider>,
 	) -> Result<(), anyhow::Error> {
-		let mut stream = Box::pin(futures::stream::pending::<()>());
-		while let Some(_) = stream.next().await {}
+		let consensus_state =
+			counterparty.query_consensus_state(None, self.consensus_state_id).await?;
+		let consensus_state = ConsensusState::decode(&mut &*consensus_state)?;
+
+		let l1_state_machine_id = StateMachineId {
+			state_id: self.l1_state_machine,
+			consensus_state_id: self.l1_consensus_state_id,
+		};
+		let mut stream = counterparty
+			.state_machine_update_notification(l1_state_machine_id.clone())
+			.await?;
+		let mut latest_height = counterparty.query_latest_height(l1_state_machine_id).await? as u64;
+		while let Some(res) = stream.next().await {
+			match res {
+				Ok(event) => match consensus_state.arbitrum_consensus_type {
+					ArbitrumConsensusType::ArbitrumOrbit => {
+						let event_height = event.latest_height;
+						let latest_event = self.latest_event(latest_height, event_height).await?;
+
+						if let Some(event) = latest_event {
+							let payload = self.fetch_arbitrum_payload(event_height, event).await?;
+							let update = ArbitrumUpdate {
+								state_machine_id: StateMachineId {
+									state_id: self.state_machine,
+									consensus_state_id: self.consensus_state_id,
+								},
+								l1_height: event_height,
+								proof: ArbitrumConsensusProof::ArbitrumOrbit(payload),
+							};
+
+							let consensus_message = ConsensusMessage {
+								consensus_proof: update.encode(),
+								consensus_state_id: self.consensus_state_id,
+								signer: counterparty.address(),
+							};
+
+							let _ = counterparty
+								.submit(
+									vec![Message::Consensus(consensus_message)],
+									counterparty.state_machine_id().state_id,
+								)
+								.await;
+
+							latest_height = event_height;
+						}
+					},
+					ArbitrumConsensusType::ArbitrumBold => {
+						let event_height = event.latest_height;
+						let latest_event =
+							self.latest_assertion_event(latest_height, event_height).await?;
```

Comment:
`start_consensus` is a long running process that should not fail on errors, use `stream::unfold` to consume the state machine_notification stream and handle errrors better, check the other clients for how we currently do this.